### PR TITLE
Renamed WRange to WDist

### PR DESCRIPTION
**Pull request by Mailaender**
We planned to do that last year. Almost last remainder of https://github.com/OpenRA/OpenRA/issues/5274.

### CODE REVIEW

File: OpenRA.Game/WDist.cs [hunk 1]
```diff
@@ -20,43 +20,43 @@ namespace OpenRA
 	/// <summary>
 	/// 1d world distance - 1024 units = 1 cell.
 	/// </summary>
-	public struct WRange : IComparable, IComparable<WRange>, IEquatable<WRange>, IScriptBindable, ILuaAdditionBinding, ILuaSubtractionBinding, ILuaEqualityBinding, ILuaTableBinding
+	public struct WDist : IComparable, IComparable<WDist>, IEquatable<WDist>, IScriptBindable, ILuaAdditionBinding, ILuaSubtractionBinding, ILuaEqualityBinding, ILuaTableBinding
 	{
 		public readonly int Range;
```
Comment (pchote) [inline OpenRA.Game/WDist.cs]: These will probably want to change too.

Comment (Mailaender) [inline OpenRA.Game/WDist.cs]: I thought this was a nice opportunity to get rid of the `WRange.Range` duplication.

Comment (pchote) [inline OpenRA.Game/WDist.cs]: If you're really bothered by it you could implement the explicit operator and do `(int)myWDist` when you need it explicitly.  `WDist.Range` makes less sense than `WRange.Range`.

Comment (pchote) [inline OpenRA.Game/WDist.cs]: `.Length` would be a potential alternative.

Comment (Mailaender) [inline OpenRA.Game/WDist.cs]: Okay. For some reason MonoDevelop had be do this conversion manually. :anger:

**Comment by penev92** [comment]
This has @Phrohdoh's :+1:  btw.

**Comment by chrisforbes** [comment]
:+1: but needs a rebase. Let's get this merged before it bitrots.

**Comment by Mailaender** [comment]
Rebased without conflicts and updated the trait documentation generator.

**Comment by penev92** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/337118cb457633756fa8b503189d48bdbab19211...28ee922f6302ec58fc4b661d516dcde9efb920a2)

**Comment by Mailaender** [comment]
Created a followup for the non-internal changes: https://github.com/OpenRA/OpenRA/issues/8691.

**Pull request merged**